ipld/blockstore/src/lib.rs

```diff
@@ -15,26 +14,28 @@ pub use block::*;
 ///
 /// The cgo blockstore adapter implements this trait.
 pub trait Blockstore {
+    type Error: std::error::Error + std::fmt::Debug + Send + Sync + 'static;
```

Comment:
This is what we started with, but it was a _huge_ pain.

1. This is incompatible with anyhow, because anyhow doesn't implement `std::error::Error`.
2. Repeating this kind of constraint wherever we want to be generic is _really_ annoying.

What if we just _don't_ constrain this? Really, only _applications_ care about enforcing constraints on this type.

Reply:
That is true, the main reason I constrained it is because `anyhow::Context` requires this, and so it gets painful there, as I can't easily define a `where` clause on `Machine::Blockstore::Error` without having to write it in a __LOT__ of places.

Reply:
Yeah, I guess that's fine. We just need to make sure we have patches for the actors before we merge.